hook/writer.go

```diff
@@ -0,0 +1,38 @@
+package hook
+
+import (
+	"io"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/sourcegraph/log"
+	"github.com/sourcegraph/log/internal/configurable"
+	"github.com/sourcegraph/log/internal/encoders"
+	"github.com/sourcegraph/log/internal/sinkcores/outputcore"
+)
+
+type writerSyncerAdapter struct{ io.Writer }
+
+func (writerSyncerAdapter) Sync() error { return nil }
+
+// Writer hooks receiver to rendered log output at level in the requested format,
+// typically one of 'json' or 'console'.
+func Writer(logger log.Logger, receiver io.Writer, level log.Level, format string) log.Logger {
```

Comment:
nit: we can create a type and `const`s for format and use it instead of `string`.

However, I can see that you had a type and changed it to string :) But I think having our own type (aliasing string) shouldn't cause any problems?

Reply:
The type was internal - let me try and move it around :)

Reply:
Took a bit of shuffling but here we go: 4373c5e501828b5f1f15f2497c920df48cffec9f